flake8diff/flake8.py

```diff
@@ -147,6 +152,22 @@ def get_color_kwargs(self, details):
             header=self.color_getter('header')(get('header')),
         )
 
+    def should_include_violation(self, violation, changed_lines):
+        """
+        Check if given violation should be included.
+
+        This method accounts for strictness modes.
+        """
+        strict_mode = self.options.get('strict_mode', 'only_lines')
+        if strict_mode == 'only_lines':
+            return violation['line_number'] in changed_lines
+        elif strict_mode == 'file':
+            return True
```

Comment:
I didn't mean to suggest you move it outside the loop, I was just asking if we could cache the option so that we wouldn't have to do the if / else dance for every violation.

If we're in strict mode, then this method will always return `True`, so we only need to check if the line number is part of the diff *iff* we're not in strict mode.